easybuild/easyblocks/generic/pythonbundle.py

```diff
@@ -134,9 +134,17 @@ def make_module_extra(self, *args, **kwargs):
         if self.multi_python:
             txt += self.module_generator.prepend_paths(EBPYTHONPREFIXES, '')
         else:
-            for pylibdir in self.all_pylibdirs:
-                if os.path.exists(os.path.join(self.installdir, pylibdir)):
+
+            # the temporary module file that is generated before installing extensions
+            # must add all subdirectories to $PYTHONPATH without checking existence,
+            # otherwise paths will be missing since nothing is there initially
+            if self.current_step == 'extensions':
+                for pylibdir in self.all_pylibdirs:
                     txt += self.module_generator.prepend_paths('PYTHONPATH', pylibdir)
+            else:
+                for pylibdir in self.all_pylibdirs:
+                    if os.path.exists(os.path.join(self.installdir, pylibdir)):
+                        txt += self.module_generator.prepend_paths('PYTHONPATH', pylibdir)
```

Comment:
Minor suggestion to reduce the amount of duplicated lines of code
```suggestion
            if self.current_step == 'extensions':
                new_pylibdirs = self.all_pylibdirs
            else:
                new_pylibdirs = [l for l in self.all_pylibdirs if os.path.exists(os.path.join(self.installdir, l))]

            for pylibdir in new_pylibdirs:
                txt += self.module_generator.prepend_paths('PYTHONPATH', pylibdir)
```